doc/source/data/performance-tips.rst

```diff
@@ -122,3 +122,49 @@ If you're seeing out of memory errors during map tasks, reducing the max block s
 Note that the number of blocks a Dataset created from ``ray.data.read_*`` contains is not fully known until all read tasks are fully executed.
 The number of blocks printed in the Dataset's string representation is initially set to the number of read tasks generated.
 To view the actual number of blocks created after block splitting, use ``len(ds.get_internal_block_refs())``, which will block until all data has been read.
+
+Improving shuffle performance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some Dataset operations require a *shuffle* operation, meaning that data is shuffled from all of the input partitions to all of the output partitions.
+These operations include ``Dataset.random_shuffle``, ``Dataset.sort`` and ``Dataset.groupby``.
+Shuffle can be challenging to scale to large data sizes and clusters, especially when the total dataset size cannot fit into memory.
+
+Starting in Ray v1.13, Datasets provides an alternative shuffle implementation known as push-based shuffle for improving large-scale performance.
+We recommend trying this out if your dataset has more than 1k partitions (input files) or 1TB of data.
+
+To try this out locally or on a cluster, you can start with the `nightly release test <https://github.com/ray-project/ray/blob/master/release/nightly_tests/dataset/sort.py>`_ that Ray runs for ``Dataset.random_shuffle`` and ``Dataset.sort``.
+To get an idea of the performance you can expect, here are some run time results for ``Dataset.random_shuffle`` on 1-10TB of data on 20 machines (m5.4xlarge instances on AWS EC2, each with 16 vCPUs, 64GB RAM).
+
+.. image:: https://docs.google.com/spreadsheets/d/e/2PACX-1vQvBWpdxHsW0-loasJsBpdarAixb7rjoo-lTgikghfCeKPQtjQDDo2fY51Yc1B6k_S4bnYEoChmFrH2/pubchart?oid=598567373&format=image
```

Comment:
should we compare the push based vs the default one in the graph to demonstrate it's more appealing.?

Reply:
I tried to run the default one but it failed to complete heh... I could maybe run it on a smaller data size? But we won't be able to provide reliable comparison points past 1TB.

Reply:
ah it's tricky then